packages/nx/src/nx-cloud/utilities/git-info.ts

```diff
@@ -0,0 +1,41 @@
+import { execSync } from 'child_process';
+
+export function getGithubSlugOrNull(): string | null {
+  const gitRemote = execSync('git remote -v').toString();
+  return extractUserAndRepoFromGitHubUrl(gitRemote);
+}
+
+function extractUserAndRepoFromGitHubUrl(gitRemotes: string): string | null {
```

Comment:
A unit test to show this working would be good.

Reply:
Also, if we only support `origin` as remote name, should it be `git config --get remote.origin.url`?

Reply:
We don't really only support origin. Here's the thing.

If a user has `origin` set up and the `origin` is a github URL, then we'll use that. If the user does not have a remote named `origin`, then we pick the first remote in the list that has a github URL. 

Even if we do pick `origin` or in the absence of `origin` we pick the first remote with github, it still does not guarantee that it's the remote they want to set up. they could be working with multiple forks, or custom named remotes, etc. So we either pick origin or just one, and if it's the wrong one, then they'll have to change the URL themselves.

We could list their remotes and tell them "pick the one you want" but I think it's an overkill?

Reply:
(or `git remote get-url origin` :P )

Reply:
I'm adding a unit test now!